x/swaprouter/router_test.go

```diff
@@ -161,30 +539,79 @@ func (suite *KeeperTestSuite) TestMultihopSwapExactAmountOut() {
 	for _, tc := range tests {
 		suite.Run(tc.name, func() {
 			suite.SetupTest()
-
+			gammKeeper := suite.App.GAMMKeeper
 			swaprouterKeeper := suite.App.SwapRouterKeeper
 
-			suite.createBalancerPoolsFromCoins(tc.poolCoins)
+			suite.createBalancerPoolsFromCoinsWithSwapFee(tc.poolCoins, tc.poolFee)
 
-			// if we expect a reduced fee to apply, we set both pools in DistrInfo to replicate it being an incentivized pool
-			// each pool has three gauges, hence 6 gauges for 2 pools
-			if tc.reducedFeeApplied {
-				test := poolincentivestypes.DistrInfo{
-					TotalWeight: sdk.NewInt(2),
-					Records: []poolincentivestypes.DistrRecord{
-						{GaugeId: 1, Weight: sdk.OneInt()}, {GaugeId: 2, Weight: sdk.OneInt()}, {GaugeId: 3, Weight: sdk.OneInt()},
-						{GaugeId: 4, Weight: sdk.OneInt()}, {GaugeId: 5, Weight: sdk.OneInt()}, {GaugeId: 6, Weight: sdk.OneInt()}},
+			// if test specifies incentivized gauges, set them here
+			if len(tc.incentivizedGauges) > 0 {
```

Comment:
Can this be moved into a helper method defined on the suite and shared between tests?